### DISCUSSION TITLE
Code running twice

### DISCUSSION THREAD

**Discussion by MrTinkerman** [micropython/micropython#12657] [General]
Hi all,

My code is running twice. I did search and found a lot, but no definite answer.
This is so weird and I'm curious about what is happening.

I'm on win 10 using Thonny 4.1.3 on a XIAO ESP32-C3

My code:
`print('hello')`

The result is below, it runs twice, why?

Thanks.

SOLVED+++++++++++++++++++++++++++++++++++++++++++++++++
As I was making this post I did a final search and got this from Jul on StackOverflow:
`on thonny click run- configure interpreter on that menu there will 
be a check box at the bottom (restart before running script) 
Untick the box`

I decided to post anyway to possibly prevent headaches for folks running into this issue

![twice](https://github.com/micropython/micropython/assets/16565076/c773ea67-ba8b-4f25-ad05-206fb0045a20)

**Comment by peterhinch** [comment]
Just FYI user code should not normally be put in `boot.py`. I don't use Thonny but from your description it sounds as if it's designed to run code in named files. If so, I would also avoid using `main.py`.

**Reply by MrTinkerman** [reply]
Thanks for the tip, I'll look into it.
But I'm curious, what IDE do you use for your uP programming?

**Reply by peterhinch** [reply]
I don't use an IDE. I use the official [mpremote tool](http://docs.micropython.org/en/latest/reference/mpremote.html) and [my own fork of Dave Hylands' rshell](https://github.com/peterhinch/rshell).

I have no doubt that Thonny is a great tool for beginners but going on queries in here it's an extra layer of code that can cause confusion. As you have discovered.

**Reply by MrTinkerman** [reply]
Ah ok, well mpremote seems a bit beyond my skill level.
I'll go and do some research on other options.

Thanks
